csrc/ir_nodes.cpp

```diff
@@ -2478,10 +2478,19 @@ IterDomain* IterDomain::resize(
         right_expansion);
   }
 
+  auto iter_type = in->getIterType();
+  auto is_symbolic = !resized_id_size->isConstInt();
+  if (symbolic) {
```

Comment:
We could do some inference in some cases. 

For example, when the expansion factors are statically known, and they are positive, unless the extent of the input domain is zero, the output should be always `Iteration`. Similarly, if `right_expanstion + left_expansion > 1`, then the output should be always `Iteration`. Also, when the input is `Broadcast` and the expansion factors are `< -1`, then the output extent is negative, which should be considered an error?

Reply:
Sounds good. I'll add that.

Reply:
I pushed a version of this logic. I'll refactor it later to run at definition time if possible. I think the options are just Iteration (if zero size output, or if output is >1 and some elements from input are used) or broadcast if output is size 1 or equivalent to `full(value=pad_value)`.